nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -804,23 +807,26 @@ def screen_matches(last: bool) -> bool:
             return False
 
         with self.nested("waiting for {} to appear on screen".format(regex)):
-            retry(screen_matches)
+            retry(screen_matches, timeout)
 
-    def wait_for_console_text(self, regex: str) -> None:
-        with self.nested("waiting for {} to appear on console".format(regex)):
-            # Buffer the console output, this is needed
-            # to match multiline regexes.
-            console = io.StringIO()
+    def wait_for_console_text(self, regex: str, timeout: int = DEFAULT_TIMEOUT) -> None:
+        # Buffer the console output, this is needed
+        # to match multiline regexes.
+        console = io.StringIO()
+
+        def check_console_text(last: bool) -> bool:
             while True:
                 try:
-                    console.write(self.last_lines.get())
+                    console.write(self.last_lines.get(timeout=1))
                 except queue.Empty:
-                    self.sleep(1)
-                    continue
-                console.seek(0)
-                matches = re.search(regex, console.read())
-                if matches is not None:
-                    return
+                    return False
+                # TODO: call run_callbacks() here
```

Comment:
I suspect this is for future work

Reply:
Yeah, currently waiting for console text that never shows up breaks the cooperative multitasking model we use for `polling_condition`. Not for this PR tho.